docs/templates.rst

```diff
@@ -640,12 +640,30 @@ iterate over containers like `dict`::
     {% endfor %}
     </dl>
 
-Note, however, that **Python dicts preserve insertion order**, which may not
-be the order you expect. If order matters, you can sort data.items() or use
-the dictsort filter. (In Python versions prior to 3.7, ``dict`` objects are
-unsorted, so you might want to either pass a sorted ``list`` of ``tuple`` s
--- or a ``collections.OrderedDict`` -- to the template, or use the `dictsort`
-filter.)
+Note that **Python 3.7+ dicts preserve insertion order**, which may not be the
+order you expect (in Python versions prior to 3.7, ``dict`` objects are
+unsorted). If order matters, you can sort ``data.items()`` or use the
+`dictsort` filter; with older Python versions you also might either pass a
+sorted ``list`` of ``tuple`` s -- or a ``collections.OrderedDict`` -- to the
+template.
+
+For example::
+
+    # using `sorted`:
+    <dl>
+    {% for key, value in sorted(my_dict.items()) %}
+        <dt>{{ key|e }}</dt>
+        <dd>{{ value|e }}</dd>
```

Comment:
I would leave out the `|e` as it is not strictly necessary. Otherwise I like the new wording a lot!

Reply:
I left the `|e` in as many other examples in the document also use that; in particular, the example immediately before it does. I would suggest a separate PR to make a pass through the docs for that pattern. Having said that, I'm totally happy to make that change here now if you want.

Reply:
I like consistency so keep it as is :D